hazelcast/src/main/java/com/hazelcast/instance/BuildInfoProvider.java

```diff
@@ -81,20 +84,42 @@ static void setJetProperties(Properties properties, BuildInfo buildInfo) {
     }
 
     private static Properties loadPropertiesFromResource(String resourceName) {
-        InputStream properties = BuildInfoProvider.class.getClassLoader().getResourceAsStream(resourceName);
+        ClassLoader classLoader = BuildInfoProvider.class.getClassLoader();
+        URL lastUrl = findLastResourceURL(resourceName, classLoader);
         Properties runtimeProperties = new Properties();
+        if (lastUrl == null) {
+            //apparently there is no resource matching this name
+            return runtimeProperties;
+        }
+        InputStream propertyStream = null;
         try {
-            if (properties != null) {
-                runtimeProperties.load(properties);
+            propertyStream = lastUrl.openStream();
+            if (propertyStream != null) {
+                runtimeProperties.load(propertyStream);
             }
         } catch (Exception ignored) {
             ignore(ignored);
         } finally {
-            closeResource(properties);
+            closeResource(propertyStream);
         }
         return runtimeProperties;
     }
 
+    private static URL findLastResourceURL(String resourceName, ClassLoader classLoader) {
```

Comment:
this is somewhat fragile. it assumes the classloader with obey the recommended order of resource iterating. 
```
    * <p> This method will first search the parent class loader for the
     * resource; if the parent is <tt>null</tt> the path of the class loader
     * built-in to the virtual machine is searched.  That failing, this method
     * will invoke {@link #findResource(String)} to find the resource.  </p>
     *
     * @param  name
     *         The resource name
     *
     * @return  A <tt>URL</tt> object for reading the resource, or
     *          <tt>null</tt> if the resource could not be found or the invoker
     *          doesn't have adequate  privileges to get the resource.
     *
     * @since  1.1
     */
    public URL getResource(String name) {
```